driver-core/src/main/com/mongodb/internal/Locks.java

```diff
@@ -28,6 +28,26 @@
  * <p>This class is not part of the public API and may be removed or changed at any time</p>
  */
 public final class Locks {
+    public static void withUninterruptibleLock(final Lock lock, final Runnable action) {
```

Comment:
I agree with this, both for consistency with the Lock interface, and also because the negative "un" increases my cognitive load when reasoning about it.  Also, I think for locking what we're saying is that we should lock uninterruptibly "by default", and only lock interruptibly when when we have a reason.  So having a simple name for the default case is a good thing.

Reply:
I also agree with this naming, but I had to make the names the way they are because the original methods that were in `Locks` (`withLock`, `checkedWithLock`) use _interruptible_ locking. So to use the naming scheme proposed by Ross (it's also consistent with the naming in the `Lock` API), I will have to rename the original `Locks.withLock`/`checkedWithLock` methods, which will affect the code that have not yet been touched either by https://github.com/mongodb/mongo-java-driver/pull/1206 or by the current PR. If this is OK, I will be glad to do the renaming.

Reply:
It's ok.  Proceed.

Reply:
Done. The rename was automatic, followed by semi-manual "import the method statically" in places where it was possible (IntelliJ does not preserve static imports when refactoring).